src/components/application_manager/src/rpc_protection_mediator_impl.cc

```diff
@@ -32,24 +33,24 @@ bool RPCProtectionMediatorImpl::IsFunctionInGroup(
 }
 
 bool RPCProtectionMediatorImpl::DoesRPCNeedEncryption(
-    const uint32_t function_id, const uint32_t app_id) const {
+    const uint32_t function_id,
+    std::shared_ptr<Application> app,
+    const uint32_t conrrelation_id,
+    const bool is_rpc_serviceSecure) {
   LOG4CXX_AUTO_TRACE(logger_);
-
   const auto& rpc_encryption_manager = policy_handler_.RPCEncryptionManager();
-  const auto app = app_manager_.application(app_id);
-  LOG4CXX_DEBUG(logger_, "app for app_id: " << app_id << " is " << app.get());
-  if (!app) {
-    LOG4CXX_ERROR(logger_, "No application found for app_id " << app_id);
+  LOG4CXX_DEBUG(logger_,
+                "Function for check is "
+                    << rpc_encryption_manager.GetPolicyFunctionName(function_id)
+                    << " conrrelation_id is " << conrrelation_id);
+
+  if (!is_rpc_serviceSecure && IsExceptionRPC(function_id)) {
     return false;
   }
+
   const auto policy_app_id = app->policy_app_id();
   const auto policy_function_id =
       rpc_encryption_manager.GetPolicyFunctionName(function_id);
```

Comment:
seems that you're doing `GetPolicyFunctionName`. can we go with one call to this function?